app/Http/Libraries/Requests/Cache/TerritoryList.php

```diff
@@ -47,4 +47,102 @@ public function generate(): array
             })->toArray()
         ];
     }
+
+    private static function generateColorAndUpdate(Guild $guild): string
+    {
+        $crc32Value = crc32(time());
+        $random = random_int(1, $crc32Value);
+        $minS = 0.5;
+        $minV = 0.75;
+
+        $rgbaColor = fromHSV(
+            $random / mt_getrandmax(),
+            $minS + (1 - $minS) * ($random / mt_getrandmax()),
+            $minV + (1 - $minV) * ($random / mt_getrandmax()),
+            1
+        );
+
+        $hex = self::rgbaToHex($rgbaColor->r, $rgbaColor->g, $rgbaColor->b);
+
+        $guild->update(['color' => $hex]);
+
+        return $hex;
+    }
+
+    private static function rgbaToHex($red, $green, $blue, $alpha = 255): string
+    {
+        $redHex = dechex($red);
+        $greenHex = dechex($green);
+        $blueHex = dechex($blue);
+        $alphaHex = dechex($alpha);
+
+        $redHex = str_pad($redHex, 2, '0', STR_PAD_LEFT);
+        $greenHex = str_pad($greenHex, 2, '0', STR_PAD_LEFT);
+        $blueHex = str_pad($blueHex, 2, '0', STR_PAD_LEFT);
+        $alphaHex = str_pad($alphaHex, 2, '0', STR_PAD_LEFT);
+
+        return '#' . $redHex . $greenHex . $blueHex . $alphaHex;
+    }
+}
+class CustomColor
+{
+    public $r;
+    public $g;
+    public $b;
+    public $a;
+
+    public function __construct($r, $g, $b, $a)
+    {
+        $this->r = $r;
+        $this->g = $g;
+        $this->b = $b;
+        $this->a = $a;
+    }
 }
+
+function clamp($value, $min, $max)
+{
+    return max($min, min($max, $value));
+}
+
+function fromHSV($h, $s, $v, $a)
```

Comment:
This is a dupe function of #28 we should agree to put these in a single place together